### DISCUSSION TITLE
ERROR [Api:StorageService] Failed to write /photos/encoded-video/.immich

### DISCUSSION THREAD

**Discussion by eagle470** [immich-app/immich#13783] [Q&A]
I'm getting this error and manually creating the folder and file is not helping. 

[Nest] 519  - 10/28/2024, 4:21:48 PM   ERROR [Api:StorageService] Failed to write /photos/encoded-video/.immich: Error: EACCES: permission denied, open '/photos/encoded-video/.immich'

This is on unraid, the container is set to run in privileged mode (as an attempted resolution.) I'm stumped.

**Comment by alextran1502** [comment]
You shouldn't create these folders manually. You should first find the fix for the permission.

Is this a brand-new install?

**Reply by eagle470** [reply]
Yes, I' have wiped and tried to reset it as cleanly as possible multiple times and it's still failing.

**Reply by eagle470** [reply]
This was an issue with a dirty database. I was attempting to use a DB from a previous install, as soon as I deleted the DB and recreated it, the install went with no issue.

I have seen reports of this issue cropping up on upgrades, not sure what would cause that.

**Reply by alextran1502** [reply]
The error message doesn't shown anything about the database being dirty though, the error code is file system permission 🤔

**Reply by RonaldPM** [reply]
This is similar to the error which I saw last day. I am not sure why but the Postgres directory which Immich create has the owner set as ollama. Is this expected or is something wrong at my end?

**Comment by rclough** [comment]
I'm also experiencing this issue since upgrading to the latest immich image on unraid. All I did was bump image versions. Not exactly keen on clearing my DB, any insight on a more targeted fix?

**Reply by rclough** [reply]
For now I've paused my docker run, but if you need info, I can resume it to get logs or filesystem info or postgres info.

Here's a snippet of the repeated logs:


```
Starting api worker
[Nest] 824  - 11/03/2024, 8:24:51 PM     LOG [Microservices:EventRepository] Initialized websocket server
[Nest] 824  - 11/03/2024, 8:24:51 PM     LOG [Microservices:MapRepository] Initializing metadata repository
[Nest] 824  - 11/03/2024, 8:24:51 PM     LOG [Microservices:MetadataService] Initialized local reverse geocoder
[Nest] 824  - 11/03/2024, 8:24:51 PM     LOG [Microservices:SystemConfigService] LogLevel=log (set via system config)
[Nest] 824  - 11/03/2024, 8:24:51 PM     LOG [Microservices:MapRepository] Initializing metadata repository
[Nest] 824  - 11/03/2024, 8:24:51 PM     LOG [Microservices:MetadataService] Initialized local reverse geocoder
[Nest] 824  - 11/03/2024, 8:24:51 PM     LOG [Microservices:ServerService] Feature Flags: {
  "smartSearch": false,
  "facialRecognition": false,
  "duplicateDetection": false,
  "map": true,
  "reverseGeocoding": true,
  "importFaces": false,
  "sidecar": true,
  "search": true,
  "trash": true,
  "oauth": false,
  "oauthAutoLaunch": false,
  "passwordLogin": true,
  "configFile": false,
  "email": false
}
[Nest] 824  - 11/03/2024, 8:24:51 PM     LOG [Microservices:StorageService] Verifying system mount folder checks (enabled=false)
[Nest] 824  - 11/03/2024, 8:24:51 PM     LOG [Microservices:StorageService] Writing initial mount file for the encoded-video folder
[Nest] 824  - 11/03/2024, 8:24:51 PM    WARN [Microservices:StorageService] Found existing mount file, skipping creation
[Nest] 824  - 11/03/2024, 8:24:51 PM   ERROR [Microservices:StorageService] Failed to write /photos/encoded-video/.immich: Error: EACCES: permission denied, open '/photos/encoded-video/.immich'
microservices worker error: Error: Failed to write "<UPLOAD_LOCATION>/encoded-video/.immich - Please see https://immich.app/docs/administration/system-integrity#folder-checks for more information."
microservices worker exited with code 1
```

**Reply by rclough** [reply]
some filesystem info:

```
root@226dfc64ddfc:/# ls -lahtr /photos
total 0
drwxr-xr-x 1 abc  users  10 Aug  5 20:42 library
drwxrwxrwx 1 abc  users  72 Aug  5 20:52 .
drwxr-xr-x 1  911  1001  58 Aug  5 20:52 thumbs
drwxr-xr-x 1  911  1001  77 Nov  3 20:23 encoded-video
drwxr-xr-x 1 root root  374 Nov  3 20:23 ..
root@226dfc64ddfc:/# ls -lahtr /photos/encoded-video/
total 16K
drwxrwxrwx 1 abc users   72 Aug  5 20:52 ..
drwxr-xr-x 1 911  1001 8.0K Aug  5 23:41 b6291e2e-5d20-4bbd-9118-12dc82f890e3
drwxr-xr-x 1 911  1001   77 Nov  3 20:23 .
-rw-r--r-- 1 911  1001   13 Nov  3 20:23 .immich
```

**Reply by rclough** [reply]
ok, checking `top`, it seems immich is run by user `abc`, I'm guessing an older version of the image had a different user, or something about the underlying system changed making the users be interpreted differently?

**Reply by rclough** [reply]
update: based on the above, I ran the following:

```
chown -R abc:users /photos/encoded-video
```

which then shifted the error, somewhat predictably, to the thumbs folder, where I repeated the process (takes much longer):

```
chown -R abc:users /photos/thumbs
```

**Reply by rclough** [reply]
@alextran1502 just tagging you in case this info helps future people, and in case you're aware of changes to the Dockerfile that may have caused this, as to hopefully prevent it from reoccurring in the future

**Reply by rclough** [reply]
actually, just checked and my image seems to come from here, so I'll move reports over there: https://github.com/imagegenius/docker-immich/issues

**Comment by didi767** [comment]
Hi,

I'm literally going insane, been trying to troubleshoot this for the past 3 hours :)
it worked perfectly until I decided to upgrade to the latest version.
I made sure that the mount permissions are the right ones (they were because nothing has changed, the only thing I did is to upgrade the immich version, nothing beyond that)
I get the same issue. I checked and it seems like I can touch to that path and see the .immich file on the windows mount.
why is it still showing me this error message:

immich_server            | [Nest] 6  - 01/11/2025, 1:25:05 AM   ERROR [Microservices:StorageService] Failed to write upload/library/.immich: Error: EACCES: permission denied, open 'upload/library/.immich'

what's going on, this is super frustrating

**Reply by coltography** [reply]
Did you ever figure this out? This is why I wait so long to update immich and low and behold the entire things been broken by this small issue that has no easy fix.

**Reply by RavingGrob** [reply]
I am also going through a similar issue, although I _**am**_ using the NixOS Module to install it.

The module works as expected when using the default mediaLocation "/var/lib/immich".

But as soon as I try to move that to another location (the partition my home directory is in, is much larger) it shoots out this error, and no amount of permission changing makes any difference.

It's taken me a few days of troubleshooting, but I have gotten to the point that the user "immich" can now ```touch``` files in the specified directory, but it hasn't impacted the ability for immich to create the files and folders it needs there.

**Reply by busybox11** [reply]
As a workaround, instead of using `services.immich.mediaLocation`, I have bound the target directory from my other drive to `/var/lib/immich`, like this:

```nix
{
  services.immich.enable = true;

  fileSystems."/var/lib/immich" = {
    device = "/mnt/hdd/services/immich";
    options = [ "bind" ];
    neededForBoot = true; # if necessary
  };
}
```

**Reply by RavingGrob** [reply]
> As a workaround, instead of using `services.immich.mediaLocation`, I have bound the target directory from my other drive to `/var/lib/immich`, like this:

Getting this working tonight wasn't on my bingo card. Thank you so much for that. Instantly worked!

**Comment by Barrow1990** [comment]
I seem to have an unusual setup for immich.
My folder structure is:

Photos / Libraries / {all my personal photos } (this is actually a remote share
Photos / Photos / {All the config folders | backups, encoded-video, library, profile, thumbs and upload) This is also a remote but located somewhere else

I recently did a restore and I had the above issue.
What I needed to do was to remove all the .immich files from each of the config folders. Then I had to exec into the docker container and create the files and then `chown abc:users {folder}` This resulted in the backup working and being able to re-sync back to where I was previously when the backup was taken.

**Comment by Jay-GitnStuff** [comment]
Over a year and this is still an issue. I'm trying to set up for the first time. I'm getting the same permission issues everyone else seems to have. My upload location folder is a mounted share from TrueNAS. I can set the user to whatever I want but nothing works. I did set the upload location folder to a local location and let Immich create it to see what user and group owned it. It was root:root. I tried mimicking the DB folder user and group, I did try root:root, but the only way I can get the container to start up is to set the location folder to full access (777), which I don't want to do for obvious reason.

Has anyone really figured this out? How do I need to set up this share?

**Reply by sirgiftzwerg** [reply]
Hi, this approach helped me get rid of the permission error (with Ubuntu + Docker Desktop trying to upgrade from 1.131.1 to release): 

Stop the Immich container, then:
Create all required subfolders
Set ownership recursively to UID 99 / GID 100
Set directory permissions so UID 99 can write + traverse 

Example (host paths will vary):

```
docker compose stop immich

mkdir -p immich_photos/{library,encoded-video,thumbs,profile,upload,backups}
chown -R 99:100 immich_photos
find immich_photos -type d -exec chmod 770 {} \;
find immich_photos -type f -exec chmod 660 {} \;

docker compose up -d immich
```


This is what ChatGPT came up with and concluded: 

Immich checks several storage folders at startup, not just one.
It tries to write a small .immich file into each folder (library, encoded-video, thumbs, etc.). If any one of them isn’t writable, Immich immediately exits. That’s why the error seemed to “move” from one folder to another.

At the same time, Immich runs as a non-root user inside the container (UID 99 / GID 100). So the folders on the host must be writable for that user, not just owned by root or your normal Linux user. Even one folder with missing permissions or missing execute (+x) access will cause startup to fail.
